lib/clientv2/node.go

```diff
@@ -59,6 +63,65 @@ func (r nodes) Update(ctx context.Context, res *apiv2.Node, opts options.SetOpti
 
 // Delete takes name of the Node and deletes it. Returns an error if one occurs.
 func (r nodes) Delete(ctx context.Context, name string, opts options.DeleteOptions) (*apiv2.Node, error) {
+	pname, err := names.WorkloadEndpointIdentifiers{Node: name}.CalculateWorkloadEndpointName(true)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get all weps belonging to the node
+	weps, err := r.client.WorkloadEndpoints().List(ctx, options.ListOptions{
+		Prefix: true,
+		Name: pname,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Collate all IPs across all endpoints, and then release those IPs.
+	ips := []net.IP{}
+	for _, wep := range weps.Items {
+		for _, ip := range wep.Spec.IPNetworks {
+			ips = append(ips, *net.ParseIP(ip))
+		}
+	}
+	_, err = r.client.IPAM().ReleaseIPs(context.Background(), ips)
+	fmt.Printf("Releasing: %v", ips)
```

Comment:
Should update to use the logger (if actually needed - does the IPAM code log this out?)

Reply:
same with other `fmt.Printf` calls in this file

Reply:
Oops, leftover debug statements.